### ISSUE TITLE
analyze: systemd-analyze doesn't report kernel time

### ISSUE THREAD

**Issue by Shuangistan** [systemd/systemd#7721]
### Submission type

 - Bug report

### systemd version the issue has been seen with

> v230, checked no significant changes of code in v236 

### Used distribution

> custom embedded

### Observation
![systemd_7721](https://user-images.githubusercontent.com/12132949/36551668-6db00c04-17f8-11e8-999c-8fd066534c2b.png)

`systemd-analyze` doesn't report the kernel time:
`Startup finished in 1.860s (userspace) = 5.957s`

But this is reported in journal by `manager.c`:
`systemd[1]: Startup finished in 4.097s (kernel) + 1.860s (userspace) = 5.957s.`

`gdbus introspect -y -d org.freedesktop.systemd1 -o /org/freedesktop/systemd1 | grep Kernel` returns
```
readonly t KernelTimestamp = 0;
readonly t KernelTimestampMonotonic = 0;
```

At the main.c, directly after `dual_timestamp_from_monotonic(&kernel_timestamp, 0);`, `kernel_timestamp.realtime` is 0

### Expectation
I am not sure if in this case the kernel time shall not be reported by `systemd-analyze`.
Also, the report from `systemd-analyze` and `manager.c` is not consistent. (see above)

**Comment by yuwata** [comment]
My system outputs:
```
$ systemd-analyze 
Startup finished in 1.745s (firmware) + 8.534s (loader) + 875ms (kernel) + 628ms (initrd) + 3.497s (userspace) = 15.281s
graphical.target reached after 3.488s in userspace
```
and journal log contains
```
Dec 25 10:33:57 systemd[1]: Startup finished in 875ms (kernel) + 628ms (initrd) + 3.497s (userspace) = 15.281s.
```
Also,
```
$ gdbus introspect -y -d org.freedesktop.systemd1 -o /org/freedesktop/systemd1 | grep Kernel
      readonly t KernelTimestamp = 1514165632747214;
      readonly t KernelTimestampMonotonic = 0;
```

**Comment by yuwata** [comment]
I've created PR #7738 for this. Please take a look.

**Comment by Shuangistan** [comment]
I'll check if it's solve my issue.

**Comment by Shuangistan** [comment]
Sorry for the delay. In-between I was stuck by other tasks.
Add firmware and loader time to manager is nice. But this does not solve all my strange problem:

systemd-analyze doesn't report the kernel time:
`$ systemd-analyze`
`Startup finished in 1.860s (userspace) = 5.957s`

That means in analyze.c: `t->kernel` returns 0:
`        if (t->kernel_time)`
`                size = strpcpyf(&ptr, size, "%s (kernel) + ", format_timespan(ts, sizeof(ts), t->kernel_done_time, USEC_PER_MSEC));`

But the value is available in manager.c. In other word, analyze.c cannot get kernel_time.

**Comment by yuwata** [comment]
@Shuangistan Thank you for testing #7738. And sorry that I may misunderstand the point. I've dropped the reference to this in the commit.

**Comment by keszybz** [comment]
@Shuangistan can you reproduce the issue with current master, and in particular with PR #7738? In your original report you said that you're using systemd-230, and there have been many fixes and changes since then.

**Comment by Shuangistan** [comment]
I'll check and try. This only happens on an arm64 cpu.

**Comment by Shuangistan** [comment]
Shall be fixed in #9790.

**Comment by kmarinushkin** [comment]
Based on my tests, PR https://github.com/systemd/systemd/pull/9801 fixes this issue.
